File: 108.Convert-SortedArraytoBinarySearchTree/clywin123/sortedArrayToBST.py
class Solution:
    # @param {integer[]} nums
    # @return {TreeNode}
    def sortedArrayToBST(self, nums):
        if(len(nums)==0):
            return 
        middle = len(nums)/2
        node = TreeNode(nums[middle])
        if(middle-1>=0):
            node.left = self.sortedArrayToBST(nums[0:middle])
        if(middle+1<len(nums)):
            node.right = self.sortedArrayToBST(nums[middle+1:len(nums)])
        return node
        

File: 108.Convert-SortedArraytoBinarySearchTree/wyrj/sortedArrayToBST.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {number[]} nums
 * @return {TreeNode}
 */
var sortedArrayToBST = function(nums) {
    const f = (leftIndex, rightIndex) => {
        if (leftIndex > rightIndex) return null;
        const mid = Math.ceil((leftIndex + rightIndex) / 2);
        const root = new TreeNode(nums[mid]);
        root.left = f(leftIndex, mid - 1);
        root.right = f(mid + 1, rightIndex);
        return root;
    }
    return f(0, nums.length - 1);
};
